pkg/snapshot/clonetree.go

```diff
@@ -202,3 +203,55 @@ func copyLink(path, targetPath string) error {
 
 	return nil
 }
+
+// cloneContentByClass copies all the content related to a given device class (devClass), possibly filtering out devices whose name
+// does NOT pass a filter (filterName). Each entry in `/sys/class/$CLASS` is actually a symbolic link. We can filter out entries depending
+// on the link target. Each filter is a simple function which takes the entry name or the link target and must return true if the
+// entry should be collected, false otherwise. last but not list, explicitely collect a list of attributes for each entry, given
+// as list of glob patterns as `subEntries`. Return the final list of glob patterns to be collected.
+func cloneContentByClass(devClass string, subEntries []string, filterName func(string) bool, filterLink func(string) bool) []string {
```

Comment:
suggestion: make a typedef for the filter function:

```go
type FilterFunc func(string) bool
```

and then use that typedef in this function signature:

```go
func cloneContentByClass(
    devClass string,
    subEntries []string,
    filterName FilterFunc,
    filterLink FilterFunc,
) []string {
...
}
```

Reply:
ack, done